__tests__/app.test.js

```diff
@@ -28,4 +28,38 @@ describe("/api/topics", () => {
   });
 });
 
+describe("/api/articles/:article_id", () => {
+  test("GET:200 sends an article object with the correct properties to the client", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.article).toMatchObject({
+          author: "butter_bridge",
+          title: "Living in the shadow of a great man",
+          article_id: 1,
+          body: "I find this existence challenging",
+          topic: "mitch",
+          votes: 100,
+        });
+      });
+  });
+  test("GET:404 sends an error message when given a valid but non-existent id", () => {
+    return request(app)
+      .get("/api/articles/999")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("No Article exists with that Id!");
+      });
+  });
+  test("GET:400 sends an error message when given an invalid id", () => {
+    return request(app)
+      .get("/api/articles/not_a_number")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid Id, Id must be a number!");
```

Comment:
lovely test, but in some instances you may find that this is too specific of an error message, you may find that it may well not always have to be a number.

Reply:
changed it to a more generic "Invalid id"!